### DISCUSSION TITLE
Using convert_names_case work for arguments default values

### DISCUSSION THREAD

**Discussion by keshe4ka** [mirumee/ariadne#1082] [Q&A]
I used to use the `@convert_kwargs_to_snake_case`, which is now marked as deprecated. Instead, it is suggested to use the `convert_schema_names` parameter in `make_executable_schema` (I think the `convert_names_case` parameter was meant).

Here is the code for creating the schema:
```
schema = make_executable_schema(
  type_defs,
  convert_names_case=True
)
```

However, problems arise when I use a nested input type, for example like this:
```
getUsers(  
    filters: UsersFilter = {  
      userId: null,   
      limit: null,   
      offset: null,   
      orderBy: {dateOfCreation: desc}  
    }  
): UsersResult!
```
For this query, I have the following resolver:
```
@token_required()
def get_users_resolver(obj, info, filters) -> dict:
  ...
  print(filters)
  ...
```

But in this case, I get the following output:
`{'user_id': None, 'limit': None, 'offset': None, 'order_by': {'dateOfCreation': 'desc'}}`

This suggests that nested arguments were not converted.
I wanted to create an issue about this, but I hesitated whether I use the option correctly, maybe I need to add something else?

`ariadne 0.19.1`

**Comment by rafalp** [comment]
Hi!

~~`convert_names_case` is not used because this feature enables for greater customization than just converting the case of names between schema and Python world.~~ Nevermind, this is not my day 😅

I need to see how `dateOfCreation` is defined in schema. New feature scans entire schema and sets new names for all objects I've been familiar with. I don't see a reason why it would skip the `out_names` for nested inputs.

**Reply by keshe4ka** [reply]
Here is the complete schema code:
```
type Query{
    getUsers(filters: UsersFilter): UsersResult!
}

input UsersFilter{
    userId: Int = null
    limit: Int = null
    offset: Int = null
    orderBy: UserOrderFields = {dateOfCreation: desc}
}

input UserOrderFields{
    name: Sort
    dateOfCreation: Sort
}

enum Sort{
    asc
    desc
}
```

**Comment by rafalp** [comment]
`convert_names_case` converts nested parameters:

```
{'filters': {'user_id': 123, 'limit': None, 'offset': None, 'order_by': {'date_of_creation': 'desc'}}}
```

But it seems that [graphql-core](https://github.com/graphql-python/graphql-core/tree/main) is not applying out names to default values.

We could do this ourselves on Ariadne side, but that would be done via schema mutation, and default value would then print using `snake_case` in GraphQL docs.

**Reply by keshe4ka** [reply]
That is, the problem concerns only the default values. However, I did not quite understand how to solve this problem other than using `@convert_kwargs_to_snake_case`
